src/OptimismMintableERC20Upgradeable.sol

```diff
@@ -0,0 +1,120 @@
+    bytes32 private constant OptimismMintableERC20StorageLocation =
+        0x6fd4c0a11d0843c68c809f0a5f29b102d54bc08a251c384d9ad17600bfa05d00;
+
+    /* STRUCTS */
+
+    /// @custom:storage-location erc7201:morpho.storage.OptimismMintableERC20
+    struct OptimismMintableERC20Storage {
+        address _remoteToken;
+        address _bridge;
+    }
+
+    /* EVENTS */
+
+    /// @dev Emitted whenever tokens are minted for an account.
+    event Mint(address indexed account, uint256 amount);
+
+    /// @dev Emitted whenever tokens are burned from an account.
+    event Burn(address indexed account, uint256 amount);
+
+    /* ERRORS */
+
+    /// @notice Reverts if the address is the zero address.
+    error ZeroAddress();
+
+    /// @notice Reverts if the caller is not the bridge.
+    error NotBridge(address caller);
+
+    /* MODIFIERS */
+
+    /// @dev A modifier that only allows the bridge to call.
+    modifier onlyBridge() {
+        OptimismMintableERC20Storage storage $ = _getOptimismMintableERC20Storage();
+        require(_msgSender() == $._bridge, NotBridge(_msgSender()));
+        _;
+    }
+
+    /* INITIALIZER */
+
+    /// @dev Sets the values for {remoteToken} and {bridge}.
+    /// @dev All two of these values are immutable: they can only be set once during initialization.
+    function __OptimismMintableERC20_init(address remoteToken_, address bridge_) internal onlyInitializing {
+        __OptimismMintableERC20_init_unchained(remoteToken_, bridge_);
+    }
+
+    function __OptimismMintableERC20_init_unchained(address remoteToken_, address bridge_) internal onlyInitializing {
+        require(remoteToken_ != address(0), ZeroAddress());
+        require(bridge_ != address(0), ZeroAddress());
+
+        OptimismMintableERC20Storage storage $ = _getOptimismMintableERC20Storage();
+        $._remoteToken = remoteToken_;
+        $._bridge = bridge_;
+    }
+
+    /* EXTERNAL */
+
+    /// @dev Allows the StandardBridge on this network to mint tokens.
+    function mint(address to, uint256 amount) external override onlyBridge {
+        _mint(to, amount);
+        emit Mint(to, amount);
+    }
+
+    /// @dev Allows the StandardBridge on this network to burn tokens.
+    function burn(address from, uint256 amount) external override onlyBridge {
+        _burn(from, amount);
+        emit Burn(from, amount);
+    }
+
+    /// @notice ERC165 interface check function.
+    /// @param _interfaceId Interface ID to check.
+    /// @return Whether or not the interface is supported by this contract.
+    function supportsInterface(bytes4 _interfaceId) external pure returns (bool) {
+        bytes4 interfaceERC165 = type(IERC165).interfaceId;
+        bytes4 interfaceOptimismMintableERC20 = type(IOptimismMintableERC20).interfaceId;
+        return _interfaceId == interfaceERC165 || _interfaceId == interfaceOptimismMintableERC20;
```

Comment:
Can't we merge this in a single mask?

Reply:
I assume you mean doing:
```solidity
bytes4 aggregatedInterface = interfaceERC165 ^ interfaceOptimismMintableERC20;
return _interfaceId == aggregatedInterface
```
Then no we can't do that because it wouldn't comply to ERC165, that specifies that `supportsInterface` has to return true on `interfaceERC165`